src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/LongPollingTransport.java

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+package com.microsoft.signalr;
+
+import io.reactivex.Completable;
+import okhttp3.Headers;
+import okhttp3.Request;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class LongPollingTransport implements Transport {
+    private OnReceiveCallBack onReceiveCallBack;
+    private TransportOnClosedCallback onClose;
+    private String url;
+    private final HttpClient client;
+    private final Map<String, String> headers;
+    private Boolean active;
+    private String pollUrl;
+    private final Logger logger = LoggerFactory.getLogger(LongPollingTransport.class);
+
+    public LongPollingTransport(Map<String, String> headers, HttpClient client) {
+        this.headers = headers;
+        this.client = client;
+    }
+
+    @Override
+    public Completable start(String url) {
+        this.active = true;
+        logger.info("Starting LongPolling transport");
+        this.url = url;
+        pollUrl = url + "&_=" + System.currentTimeMillis();
+        logger.info("Polling {}", pollUrl);
+        HttpRequest request = new HttpRequest();
+        request.addHeaders(headers);
+        HttpResponse response = this.client.get(pollUrl, request).blockingGet();
+        if (response.getStatusCode() != 200){
+            logger.error("Unexpected response code {}", response.getStatusCode());
+            this.active = false;
+            return Completable.error(new Exception("Failed to connect"));
+        } else {
+            this.active = true;
```

Comment:
Does this just throw away the content of the first poll response?

Reply:
The first response should be empty

Reply:
It would be good to validate that. If the response is non-empty, it's contents should probably go in an exception that gets raised to the user.

Reply:
This isn't something we do in any of the clients. I don't think it's necessary. But we can add it